pdns/ipcipher.cc

```diff
@@ -59,10 +59,40 @@ static ComboAddress encryptCA6(const ComboAddress& address, const std::string& k
 
   ComboAddress ret = address;
 
+#if OPENSSL_VERSION_MAJOR >= 3
+  auto ctx = std::unique_ptr<EVP_CIPHER_CTX, decltype(&EVP_CIPHER_CTX_free)>(EVP_CIPHER_CTX_new(), &EVP_CIPHER_CTX_free);
+  if (ctx == nullptr) {
+    throw pdns::OpenSSL::error("encryptCA6: Could not initialize cipher context");
+  }
+
+  EVP_CIPHER_CTX_set_padding(ctx.get(), 0);
```

Comment:
I'm not sure how to add a static assertion here. The only way I can find is to either use the `OSSL_CIPHER_PARAM_BLOCK_SIZE` parameter on the EVP context object or call `EVP_CIPHER_get_block_size`. However the documentation mentions the following:

```
"blocksize" (OSSL_CIPHER_PARAM_BLOCK_SIZE) <unsigned integer>
           Gets the block size for the associated cipher algorithm cipher.  The block size should be 1 for stream ciphers.  Note that the block size for a cipher may be different to the block
           size for the underlying encryption/decryption primitive.  For example AES in CTR mode has a block size of 1 (because it operates like a stream cipher), even though AES has a block
           size of 16.  Use EVP_CIPHER_get_block_size() to retreive the cached value.
```

Reply:
Sorry, I should have been more clear. I had in mind a static assertion checking that the size of the data we are going to encrypt is 16 bytes, with a message explaining that this is mandatory given that padding is disabled.
What I'm worried about is someone reusing that code later with some data that is not a multiple of the block size, or a different algorithm, without understanding that limitation.